fix incorrect L1 period param documentation

tstastny opened the pull request:
Minor bug - small PR.

I noticed while working on #18810 that the L1 period param has been incorrectly documented for some time now (wrong units -- should be seconds):
https://github.com/PX4/PX4-Autopilot/blob/08b0ac9654d0041a78c61025a4114a59b0045f4d/src/modules/fw_pos_control_l1/fw_pos_control_l1_params.c#L46-L61

It is correct in the L1 library:
https://github.com/PX4/PX4-Autopilot/blob/08b0ac9654d0041a78c61025a4114a59b0045f4d/src/lib/l1/ECL_L1_Pos_Controller.hpp#L220

I guess this is just left over documentation from when the L1 distance was still not dynamic with ground speed (some time ago) and this param was literally the L1 distance.

My suggested changes:
- documentation fix (correct units, etc)
- lower the minimum period and period recommendations to what an L1 distance at ground speed 15 m/s would translate to

^The latter point here open for debate if anyone has opinions (I didnt change the default).

Derivation of "5-10" seconds comes from the following equation:
```
l1_distance = ground_speed * damping * period /  pi
```
Solve for period with the previously suggested 18-25 meters (L1 distance) and you get 5-7s.

Will keep as draft until anyone else chimes in.

bresch commented:
I think a lot of user are flying with the default value of 20s, which makes a distance of ~70m @ 15m/s. I understand that if the suggested value of 18-25m is correct, we should change the period to 5-7s, but this would make the control gain much higher.
Is the current controller really way too detuned?

bresch commented:
I think the recommended value was always a "period" and never a "distance": https://github.com/PX4/PX4-Autopilot/commit/f595b204eab82679a52ca5f43408797988cfdf42#diff-f14ad8390b0107a6c83c7a328fbc0bb732dc39e108efe74ecef40e76502844c5

bresch commented on src/modules/fw_pos_control_l1/fw_pos_control_l1_params.c:
```diff
@@ -46,13 +46,13 @@
- * A value of 18-25 meters works for most aircraft. Shorten
+ * Used to determine the L1 gain and controller time constant. This parameter is
+ * proportional to the L1 distance (which points ahead of the aircraft on the path
+ * it is following). A value of 5-10 seconds works for most aircraft. Shorten
```
```suggestion
 * it is following). A value of 18-25 seconds works for most aircraft. Shorten
```

tstastny commented:
> I think the recommended value was always a "period" and never a "distance"

Ok - I mean these values work. So no reason to change it.

> Is the current controller really way too detuned?

Just depends on the aircraft roll control dynamic. E.g. our foamies are flying with period of 10 sec (easyglider -- 10 m/s cruise, believer -- 16 m/s cruise). But yeah, lower gets quite aggressive, and one needs to know what the roll time constant is to ensure stability bounds are maintained.

sfuhrer reviewed (approved):
Thanks, that's much better.

tstastny commented:
Forgot to change the min period back to 12 sec in the second commit.. rebased and squashed all 3 back into one... as we dont need 3 commits to change 3 lines of text :)

sfuhrer commented:
> Forgot to change the min period back to 12 sec in the second commit.. rebased and squashed all 3 back into one... as we dont need 3 commits to change 3 lines of text :)

I would actually reduce the min period, maybe not to 3 but to 7 or something. As it's a valid setting for some application, even if not for the normal ones.

tstastny commented:
> I would actually reduce the min period, maybe not to 3 but to 7 or something. As it's a valid setting for some application, even if not for the normal ones.

ok done.
